taichi/ir/type_factory.cpp

```diff
@@ -1,4 +1,5 @@
 #include "taichi/ir/type_factory.h"
+#include "type_factory.h"
```

Comment:
I'm not sure if this line will cause some compilers to think `"type_factory.h"` and `"taichi/ir/type_factory.h"` are different files and include it twice?

Reply:
Good point. The file will be included twice, which is fine since we have `#pragma once`. But the second line is unnecessary so I'll remove it in a future PR :-)